src/nvim/fileio.c

```diff
@@ -2879,8 +2883,21 @@ buf_write (
           }
 #endif
 
+#ifdef HAVE_READLINK
+          // In the case where the file is a symlink, we want to copy the link
```

Comment:
could declare and default `fname_res` here. That eliminates the `else` and `#else` cases below.

    char_u *fname_res = fname;  // De-symlinked filename

Reply:
Glad to hear it, I think that's way simpler. I had it that way originally but changed it because it seemed like the rest of the file was avoiding declaring anything after the start of the function. I wasn't sure if that was a code style choice or just cruft from the old C ways of doing things.